packages/pwa-kit-react-sdk/src/ssr/universal/components/with-legacy-get-props/index.js

```diff
@@ -27,16 +28,35 @@ export const withLegacyGetProps = (Wrapped) => {
             const {params} = match
 
             const components = [App, route.component]
-            const promises = components.map((c) =>
-                c.getProps
-                    ? c.getProps({
-                          req,
-                          res,
-                          params,
-                          location
-                      })
+            const promises = components.map((c, i) => {
+                // getTemplateName is a promise and it's intentially not awaited here
+                // to avoid blocking the execution of the getProps function to maximize performance
+                // getTemplateName should be very fast, under 0.2ms
+                c.getTemplateName().then((templateName) => {
+                    res.__performanceTimer.mark(
+                        `${PERFORMANCE_MARKS.getProps}::${templateName}`,
+                        'start'
+                    )
+                })
```

Comment:
I like the idea that we aren't awaiting on that function to get the template name as the performance of the rendering should be paramount. But we have to be aware that on this minuscule level, we are effecting the true performance time of getProps.

Reply:
Yea, this is a decision that i want to make, I'd be rather to be a tiny bit inaccurate than having it to impact performance.

I measured the time for `c.getTemplateName()` to resolve is 0.2ms, but I want to be on the safe side in case there are some cpu blips that cause it to take longer.